wasmcloud_host/README.md

```diff
@@ -1,19 +1,172 @@
+
+    ```console
+    $ fnm use 14
+    Using Node v14.17.6
+    ```
+
+1. Clone the repo `wasmCloud/wasmcloud-otp` into some suitable development directory
+
+    ```console
+    $ git clone https://github.com/wasmCloud/wasmcloud-otp.git
+    ```
+
+1. Install and start `nats-server` using the JetStream option
+
+    ```console
+    $ brew install nats-server
+    $ nats-server -js
+    ```
+
+This terminal window is now occupied with the `nats-server` console logs
+
+## Local Server Build
+
+In a new terminal, change into the `wasmcloud-otp` folder, then change into the `assets` folder underneath `wasmcloud_host`
+
+1. `cd wasmCloud/wasmcloud-otp/wasmcloud_host`
+
+1. Run `npm install`.  As long as you have switched to version `14.xx.yy` of NodeJs, this should run successfully
+
+1. Change up one directory level and run `make run-interactive`.  This will build the wasmcloud server and connect it to your already-started NATS server
+
+1. This terminal window now shows both an Elixir command prompt and from time to time, any wasmcloud console logs
+
+1. You can now point your browser to (localhost:4000)[localhost:4000] to see the wasmcloud dashboard
+
+## Building Capability Providers
+
+In the `capability_providers` folder, there are two simple capability providers: `httpserver` and `keyvalue`
+
+Both providers must be built using the supplied `Makefile`, so first change into the `capability_providers` directory
+
+### Building `httpserver`
+
+The `httpserver` provider has been written in Go and can be built using the command:
+
+```console
+$ make httpserver-<os>
+```
+
+Where `<os>` is either `linux`, `mac` or `windows`
+
+E.G. To build the Mac `httpserver` provider
+
+```console
+$ make httpserver-mac
+mkdir -p ./_build
+cd httpserver/go && \
+        GOOS=darwin go build -o httpserver-mac && \
+        mv httpserver-mac ../../_build/httpserver-mac
+```
+
+## Building `keyvalue`
+
+The `keyvalue` provider has been written in Rust and ***not*** is built not using `cargo`!
+
+Due to the fact that different compilation targets can be generated by the same `Makefile`, the Rust cross compiler `cross` is used instead of `cargo`.
+
+1. Install (or update) `cross`
+
+    ```console
+    $ cargo install cross
+    Updating crates.io index
+      Downloaded cross v0.2.1
+      Downloaded 1 crate (175.1 KB) in 0.27s
+      Installing cross v0.2.1
+      Downloaded cc v1.0.70
+      Downloaded which v3.1.1
+
+      <snip>
+
+      Downloaded libc v0.2.101
+      Downloaded 12 crates (1.2 MB) in 0.48s
+       Compiling libc v0.2.101
+
+      <snip>
+
+      Compiling toml v0.5.8
+       Finished release [optimized] target(s) in 40.28s
+     Installing /Users/chris/.cargo/bin/cross
+      Installed package `cross v0.2.1` (executable `cross`)
+    ```
+
+1. Make sure that `cross` is used instead of `cargo`
+
+    ```console
+    $ export CARGO=CROSS
+    ```
+
+1. Create the `keyvalue` provider
+
+    ```console
+    $ make keyvalue-mac
```

Comment:
I much prefer to do it inline, so you don't pollute your shell's environment:

```suggestion
    $ CARGO=cross make keyvalue-mac
```
or
```suggestion
    $ make keyvalue-mac CARGO=cross 
```

In practice, I don't think `make keyvalue-mac` needs cross: only the par targets?

Reply:
In code blocks, especially where users will probably want to cut and paste several lines, I think it makes sense to drop the `$` prompt as it really just gets in the way